lib/Common/sensors_state.h

```diff
@@ -5,6 +5,7 @@
 struct SensorState {
   bool brewSwitchState;
   bool steamSwitchState;
+  bool dreamSteamSwitchState;
```

Comment:
is this going to be based off a physical switch to enable/disable DreamSteam? why not make it a toggle in the UI, as a system-wide preference?

Reply:
The issue is that the ui is limited in how many more global components can be added, i think basing this off a physical switch might be the only option for now, alternatively it can be made global and scrapped later if there is anything more important to replace it on the horizon 🤷

Reply:
> is this going to be based off a physical switch to enable/disable DreamSteam? 

Yes, the intention was to use it with a physical switch. If there's no separate switch then the value shall be based on the macro state.

Reply:
took the liberty to build on top of the logic changes using a UI toggle and storing the setting in eeprom, proposed changes in https://github.com/Zer0-bit/gaggiuino/pull/463.

unless the physical toggle serves another purpose or use case, this should work universally for all installs. thoughts?

Reply:
I have moved this pr to a draft for now, as part of the new ctrl stack this will have to be revised for sure.